stores/metadata.go

```diff
@@ -671,9 +672,21 @@ GROUP BY c.fcid
 
 	sizes := make(map[types.FileContractID]api.ContractSize)
 	for _, row := range rows {
-		if types.FileContractID(row.Fcid) == (types.FileContractID{}) {
+		fcid := types.FileContractID(row.Fcid)
+		if fcid == (types.FileContractID{}) {
 			return nil, errors.New("invalid file contract id")
 		}
+
+		// adjust the amount of prunable data with the pending uploads, due to how
+		// we record contract spending a contract's size might already include
+		// pending sectors
+		pending := s.uploadingSectors.pending(fcid)
```

Comment:
Tbh I don't think `uploadingSectors` belongs into the store. It has nothing to do with persistence after all. It might be fine to not return perfectly accurate info since there is already a race here between the cached sectors and the database. e.g. a sector might not be part of the database at the time we run the query because it's uploading but the upload finishes before we call `pending` here.